internal/services/postgres/postgresql_server_resource_test.go

```diff
@@ -402,67 +406,75 @@ func (t PostgreSQLServerResource) Exists(ctx context.Context, clients *clients.C
 }
 
 func (PostgreSQLServerResource) template(data acceptance.TestData, sku, version string) string {
+	sslEnabledBlock := ``
+	if !features.FivePointOhBeta() {
+		sslEnabledBlock = `ssl_enforcement_enabled = true`
+	}
 	return fmt.Sprintf(`
 provider "azurerm" {
   features {}
 }
 
 resource "azurerm_resource_group" "test" {
-  name     = "acctestRG-psql-%d"
-  location = "%s"
+  name     = "acctestRG-psql-%[1]d"
+  location = "%[2]s"
 }
 
 resource "azurerm_postgresql_server" "test" {
-  name                = "acctest-psql-server-%d"
+  name                = "acctest-psql-server-%[3]d"
   location            = azurerm_resource_group.test.location
   resource_group_name = azurerm_resource_group.test.name
 
   administrator_login          = "acctestun"
   administrator_login_password = "H@Sh1CoR3!"
 
-  sku_name   = "%s"
-  version    = "%s"
+  sku_name   = "%[4]s"
+  version    = "%[5]s"
   storage_mb = 51200
 
-  ssl_enforcement_enabled = true
+  %[6]s
 }
-`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, sku, version)
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, sku, version, sslEnabledBlock)
 }
 
 func (r PostgreSQLServerResource) basic(data acceptance.TestData, version string) string {
 	return r.template(data, "B_Gen5_1", version)
 }
 
 func (PostgreSQLServerResource) basicWithIdentity(data acceptance.TestData, version string) string {
+	sslEnabledBlock := ``
+	if !features.FivePointOhBeta() {
+		sslEnabledBlock = `ssl_enforcement_enabled = true`
+	}
```

Comment:
Since this property is deprecated, I think it's acceptable to have one test or template that still includes it (to check that it still works) but to remove it from all other test configurations.

Since we have it in the template I'd suggest removing it from this and all other test configurations below
```suggestion
```